File: 3517/solution.py
class Solution:
    def smallestPalindrome(self, s: str) -> str:
        count = Counter(s)
        res = ''
        mid = ''
        for i in sorted(count.keys()):
            if count[i]%2:
                mid = i 
                count[i]-=1
            res+=i*(count[i]//2)
        res = res + mid + res[::-1]
        return res

File: 3517/solution.java
class Solution {
    public String smallestPalindrome(String s) {
        int n = s.length();
        int[] freq = new int[26];
        char[] res = new char[n];

        for(char c: s.toCharArray())
            freq[c-'a']++;

        int left = 0;
        int right = n-1;

        for(int i=0;i<26;i++){
            char c = (char) (i+'a');
            while(freq[i]>=2){
                res[left++] = c;
                res[right--] = c;
                freq[i]-=2;
            }
        }

        for(int i=0;i<26;i++){
            if(freq[i]==1){
                res[left] = (char) (i+'a');
                break;
            }
        }
        return new String(res);
    }
}
